app/controllers/api/v1/projects_controller.rb

```diff
@@ -66,6 +62,27 @@ def index
     super
   end
 
+  def update
+    super do |resource|
+      # TODO: extract this primary project content update
+      # to a service object that sits in the project
+      # transaction.
+      content_attributes = primary_content_attributes(update_params)
+      unless content_attributes.blank?
+        resource.primary_content.update!(content_attributes)
+      end
+
+      # TODO: move this to a service object to create/update
+      # tags for the project
+      tags = create_or_update_tags(update_params.dup)
+      resource.tags = tags unless tags.nil?
+
+      if !content_attributes.blank? || !tags.nil?
+        resource.updated_at = Time.zone.now
```

Comment:
I should probably just allow the tag belongs_to to touch the project resource here instead of manually setting the TS for the save action.